api/server/httprest/server.go

```diff
@@ -0,0 +1,105 @@
+package httprest
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/runtime"
+)
+
+var _ runtime.Service = (*Server)(nil)
+
+// httpHandler is a functional interface that implements the http handler functionality.
+type httpHandler func(
+	h http.HandlerFunc,
+	w http.ResponseWriter,
+	req *http.Request,
+)
+
+// Config parameters for setting up the http-rest service.
+type config struct {
+	httpAddr string
+	handler  httpHandler
+	router   *mux.Router
+	timeout  time.Duration
+}
+
+// Server serves HTTP traffic.
+type Server struct {
+	cfg          *config
+	server       *http.Server
+	cancel       context.CancelFunc
+	ctx          context.Context
+	startFailure error
+}
+
+// New returns a new instance of the Server.
+func New(ctx context.Context, opts ...Option) (*Server, error) {
+	g := &Server{
+		ctx: ctx,
+		cfg: &config{},
+	}
+	for _, opt := range opts {
+		if err := opt(g); err != nil {
+			return nil, err
+		}
+	}
+
+	if g.cfg.router == nil {
+		return nil, errors.New("router option not configured")
+	}
+
+	// TODO: actually use the timeout config provided
```

Comment:
What about this TODO?

Reply:
It's an existing issue, for some reason we just never had a way to provide a configuration for this via flags... probably should have something.

Reply:
Whats the deal with this? What timeout config? Why can't you use it?

Reply:
I should either remove it now or add it I guess, it's some existing config that I migrated to this implementation, but we never had a way to configure the value ( it was accepting this field previously but it was never used)

Reply:
nvm i'm just going to try to set it now